### PR TITLE
If repos is null fail quickly and do not pass null to AQL

### PR DESCRIPTION
**Pull request by shrajfr12**
The plugin uses AQL search to determine items to be deleted and if repos is NULL it is interpreted by AQL to mean all repos. Hence check if repos is null (if not passed in curl or in json or if passed with empty array) and log error message and throw CancelRequest.

### CODE REVIEW

File: cleanup/artifactCleanup/artifactCleanup.groovy [hunk 1]
```diff
@@ -159,6 +176,11 @@ if ( deprecatedConfigFile.exists() && configFile.exists() ) {
 private def artifactCleanup(String timeUnit, int timeInterval, String[] repos, log, paceTimeMS, dryRun = false, disablePropertiesSupport = false) {
     log.info "Starting artifact cleanup for repositories $repos, until $timeInterval ${timeUnit}s ago with pacing interval $paceTimeMS ms, dryrun: $dryRun, disablePropertiesSupport: $disablePropertiesSupport"
 
+    if ( !repos || (repos.length == 0) ) {
```
Comment (alexhung) [inline cleanup/artifactCleanup/artifactCleanup.groovy:179]: All these checks are identical except for the message. Can we DRY this up?

**Review by alexhung — CHANGES_REQUESTED** [review]

File: cleanup/artifactCleanup/artifactCleanup.groovy (hunk 1 above)
Comment (shrajfr12) [reply cleanup/artifactCleanup/artifactCleanup.groovy:179]: Oh they are different use cases and I am trying to minimize impacts

Comment (alexhung) [reply cleanup/artifactCleanup/artifactCleanup.groovy:179]: Different use cases? I don't understand. They are used in different place of the code, sure but the code is identical.

Comment (shrajfr12) [reply cleanup/artifactCleanup/artifactCleanup.groovy:179]: i had it in only 1 place inside artifactCleanup() but that would execute all extra stuff and fail at last minute so even though repeating the message clarifies where it gets hit

Comment (shrajfr12) [reply cleanup/artifactCleanup/artifactCleanup.groovy:179]: So your point is to wrap into a function and call each place with a different message ?

Comment (alexhung) [reply cleanup/artifactCleanup/artifactCleanup.groovy:179]: I'm still confused. Perhaps it's my lack of Groovy knowledge? I fail to see how a private function containing these code, and being called from all these locations won't work?

Comment (alexhung) [reply cleanup/artifactCleanup/artifactCleanup.groovy:179]: > So your point is to wrap into a function and call each place with a different message ?

Yes, of course. How else would one DRY these code up?

Comment (alexhung) [reply cleanup/artifactCleanup/artifactCleanup.groovy:179]: DRY = Don't Repeat Yourself, or don't create duplication.

Comment (shrajfr12) [reply cleanup/artifactCleanup/artifactCleanup.groovy:179]: Not used to that acronym!!! But it is done.. Will reverify locally and ask support person to confirm  in customer reproduction as well.

Comment (alexhung) [reply cleanup/artifactCleanup/artifactCleanup.groovy:179]: > if it is a function you want to call with a different message I can do that

Yes, that's what I meant.

File: cleanup/artifactCleanup/artifactCleanup.groovy [hunk 2]
```diff
@@ -176,11 +171,8 @@ if ( deprecatedConfigFile.exists() && configFile.exists() ) {
 private def artifactCleanup(String timeUnit, int timeInterval, String[] repos, log, paceTimeMS, dryRun = false, disablePropertiesSupport = false) {
     log.info "Starting artifact cleanup for repositories $repos, until $timeInterval ${timeUnit}s ago with pacing interval $paceTimeMS ms, dryrun: $dryRun, disablePropertiesSupport: $disablePropertiesSupport"
 
-    if ( !repos || (repos.length == 0) ) {
-        def errorMessage = 'repos parameter must be specified before initiating cleanup.'
-        log.error errorMessage
-        throw new CancelException(errorMessage, 400)
-    }
+    cancelOnNullrepos(repos,'repos parameter must be specified before initiating cleanup.' )
```
Comment (alexhung) [inline cleanup/artifactCleanup/artifactCleanup.groovy:174]: Nitpick (not blocking from merge): inconsistent formatting for `cancelOnNullrepos`:

1. If camelCase, shouldn't function name be: `cancelOnNullRepos`?
2. Some calls have trailing space before `)` some don't. None has space between args.

**Review by alexhung — APPROVED** [review]
Functional but has minor formatting/styling issues.

**Review by alexhung — APPROVED** [review]

**Pull request merged**